src/main/java/org/ballerinax/datamapper/DataMapperNodeVisitor.java

```diff
@@ -0,0 +1,185 @@
+                    String recordSignature = recordSymbol.getModule().get().id().toString();
+                    recordName = recordSignature + ":" + recordName;
+                    Map<String, RecordFieldSymbol> fieldSymbolMap = ((RecordTypeSymbol) ((TypeDefinitionSymbol)
+                            recordSymbol).typeDescriptor()).fieldDescriptors();
+                    String serialized = getFieldTypes(fieldSymbolMap);
+                    serialized = "{\"" + recordName + "\":" + serialized + "}";
+                    this.recordTypes.put(recordName, serialized);
+                }
+            }
+        }
+    }
+
+    @Override
+    public void visit(ClassDefinitionNode classDefinitionNode) {
+        NodeList<Token> classTypeQualifiers = classDefinitionNode.classTypeQualifiers();
+        for (Token classTypeQualifier : classTypeQualifiers) {
+            if (classTypeQualifier.text().equals("client")) {
+                Optional<Symbol> classSymbolOpt = model.symbol(classDefinitionNode);
+                if (classSymbolOpt.isPresent()) {
+                    Symbol classSymbol = classSymbolOpt.get();
+                    if (classSymbol.getName().isEmpty()) {
+                        continue;
+                    }
+                    String clientName = classSymbol.getName().get();
+                    Collection<MethodSymbol> methods = ((BallerinaClassSymbol) classSymbol).methods().values();
+                    Map<String, FunctionRecord> functionMap = new HashMap<>();
+                    for (MethodSymbol method : methods) {
+                        if (!method.qualifiers().contains(Qualifier.REMOTE)) {
+                            continue;
+                        }
+                        String functionName = method.getName().get();
+                        List<String> paraType = new ArrayList<>();
+                        FunctionRecord functionRecord = new FunctionRecord();
+                        for (ParameterSymbol parameter : method.typeDescriptor().parameters()) {
+                            if (parameter.getName().isPresent()) {
+                                String parameterName = parameter.getName().get();
+                                if (parameter.typeDescriptor().typeKind() == TypeDescKind.UNION) {
+                                    List<TypeSymbol> paraList = ((UnionTypeSymbol) parameter.typeDescriptor()).
+                                            memberTypeDescriptors();
+                                    for (TypeSymbol typeSymbol : paraList) {
+                                        if (typeSymbol.typeKind() == TypeDescKind.ERROR) {
+                                            continue;
+                                        } else {
+                                            if (!typeSymbol.signature().equals("()")) {
+                                                paraType.add(typeSymbol.signature());
+                                            }
```

Comment:
If you want to skip the nil type, check for the typedesc kind in the initial if condition itself. e.g.,
```java
if (typeSymbol.typeKind() == TypeDescKind.ERROR || typeSymbol.typeKind() = TypeDescKind.NIL)
```

Reply:
Addressed the comment